kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/ssl/SSLUtils.java

```diff
@@ -257,4 +257,83 @@ public static SSLEngine createSslEngine(SslContextFactory.Server sslContextFacto
         return engine;
     }
 
+    public static SSLEngine createClientSslEngine(SslContextFactory.Client sslContextFactory) throws Exception {
+        sslContextFactory.start();
+        SSLEngine engine  = sslContextFactory.newSSLEngine();
+        engine.setUseClientMode(true);
+
+        return engine;
+    }
+
+    public static SslContextFactory.Client createClientSslContextFactory(
```

Comment:
Could you reuse the code in `createSslContextFactory` instead of copying them?

Reply:
this method is returning SslContextFactory.Client and not SslContextFactory.Server 
also there are entries that should not be set like KopSslKeystoreLocation (that would mean that we use the certificate of the KOP service for client auth).
because here we are building a "client".

There is way to not copy/paste

Reply:
But they both use the same way to generate an `ImmutableMap`. I mean:

```java
    private static ImmutableMap<String, Object> buildSslConfigMap(KafkaServiceConfiguration kafkaServiceConfiguration) {
        /* ... */
    }
    public static SslContextFactory.Server createSslContextFactory(
            KafkaServiceConfiguration kafkaServiceConfiguration) {
        return createSslContextFactory(buildSslConfigMap(kafkaServiceConfiguration));
    }

    public static SslContextFactory.Client createClientSslContextFactory(
        return createClientSslContextFactory(buildSslConfigMap(kafkaServiceConfiguration));
    }
```